python/cudf/cudf/core/mixins/mixin_factory.py

```diff
@@ -93,49 +104,46 @@ def _create_delegating_mixin(
     >>> print(bar.foo1.__doc__)
     Perform the operation foo1, which returns 42.
     """
-
     docstring_attr = f"{category_name}_DOCSTRINGS"
     validity_attr = f"_VALID_{category_name}S"
 
-    class OperationMixin:
-        @classmethod
-        def _add_operation(cls, operation):
-            # This function creates operations on-the-fly.
-
-            # Generate a signature without the `op` parameter.
-            base_operation = getattr(cls, operation_name)
-            base_params = inspect.signature(base_operation).parameters.values()
-            params = [p for p in base_params if p.name != "op"]
-            signature = inspect.Signature(parameters=params)
-
-            # Generate the list of arguments forwarded to the base operation.
-            arglist = ", ".join(
-                [
-                    f"{key}={key}"
-                    for key in signature.parameters
-                    if key not in ("self", "args", "kwargs")
-                ]
-            )
-            arglist = arglist + (", " if arglist else "") + "*args, **kwargs"
-
-            # Apply the formatted docstring of the base operation to the
-            # operation being created here.
-            docstring = base_operation.__doc__.format(
-                cls=cls.__name__,
-                op=operation,
-                **getattr(cls, docstring_attr, {}).get(operation, {}),
-            )
+    class Operation:
+        """
+        """
+
+        def __init__(self):
+            self._operation_name = operation_name
 
-            namespace = {}
-            exec_str = f"""
-def {operation}{str(signature)}:
-    '''{docstring}
-    '''
-    return self.{operation_name}(op="{operation}", {arglist})
-"""
-            exec(exec_str, namespace)
-            setattr(cls, operation, namespace[operation])
+        def __set_name__(self, owner, name):
+            self._name = name
```

Comment:
This is the name of an instance of this descriptor, e.g., `"max"` or `"min"`.

Reply:
This basically replaces the need to manually override `__name__` right?

Reply:
Ah wait no, we still need to assign it to the partial. In that case, what's the benefit of defining `__set_name__` here?

Reply:
Since you're forced to call `__set_name__` manually in this case, maybe it would be cleaner to just pass the name in the constructor? I'm not familiar enough with this hook to understand the benefits here, although this may be the canonical way to set the name of a descriptor.

Reply:
Yup, there's no benefit to it here - I went with just passing `name` to the ctor.